sporkfish/searcher.py

```diff
@@ -348,6 +350,59 @@ def _negamax_lazy_smp(
             else:
                 continue  # Continue the loop if no result is ready yet
 
+    # Aspiriation windows
+    # Assumption (*): we expect the search score from depth n should be somewhat similar depth n + 1.
+    # Method: set alpha = prev_depth_score - constant, beta = prev_depth_score + constant.
+    # Do a search with narrower window [alpha, beta] as opposed to [-inf, inf].
+    # This should cause cutoffs (if they do happen) to happen faster.
+    # If the score falls within the alpha/beta bounds, our assumption (*) holds - so take that score.
+    # If the score falls outside the alpha/beta windows, the score changed too much
+    # i.e. our assumption (*) is false - so we do a full search to be safe.
+    def _aspiration_windows_search(
+        self,
+        search_func: Callable[[Board, int, float, float], Tuple[float, chess.Move]],
+        board_to_search: Board,
+        depth: int,
+        prev_score: float,
+    ) -> Tuple[float, chess.Move]:
+        """
+        Perform an aspiration windows search.
+
+        Aspiration windows are used to optimize the search process by narrowing the search window based on
+        previous search results.
+
+        :param search_func: A callable function representing the search algorithm.
+        :type search_func: Callable[[Board, int, float, float], Tuple[float, chess.Move]]
+        :param board_to_search: The chess board to search.
+        :type board_to_search: Board
+        :param depth: The search depth.
+        :type depth: int
+        :param prev_score: The score of the previous depth in a iterative deepening search.
+        :type prev_score: float
+
+        :return: A tuple containing the score and the best move found during the search.
+        :rtype: Tuple[float, chess.Move]
+        """
+        if self._config.enable_aspiration_windows and depth > 1:
+            # Aspiration window size: 50 centipawn is worth about 1/2 of a pawn in our eval function
+            # We leave configuration for this to another PR
+            window_size = 50
```

Comment:
According to your [reference](https://www.chessprogramming.org/Aspiration_Windows) - "Typical window sizes are 1/2 to 1/4 of a pawn on either side of the guess.".

I am struggling to understand how that corresponds to the `window_size` of 50 you are using here. Can you elaborate the unit here?

[edit] Clearly I was blind, just read your comment. All clear thanks